src/topology/schema.rs

```diff
@@ -8,103 +7,67 @@ use crate::{
     topology,
 };
 
-/// Create a new [`Definition`] by recursively merging all provided inputs into a given component.
-///
-/// Recursion happens when one of the components inputs references a transform that has no
-/// definition output of its own, in such a case, the definition output becomes the merged output
-/// of that transform's inputs.
-///
-/// For example:
-///
-/// Source 1 [Definition 1] ->
-/// Source 2 [Definition 2] -> Transform 1 []             -> [Definition 1 & 2]
-/// Source 3 [Definition 3] -> Transform 2 [Definition 4] -> [Definition 4]     -> Sink
-///
-/// When asking for the merged definition feeding into `Sink`, `Transform 1` returns no definition
-/// of its own, when asking for its schema definition. In this case the `merged_definition` method
-/// recurses further back towards `Source 1` and `Source 2`, merging the two into a new definition
-/// (marked as `[Definition 1 & 2]` above).
-///
-/// It then asks for the definition of `Transform 2`, which *does* defines its own definition,
-/// named `Definition 4`, which overrides `Definition 3` feeding into `Transform 2`. In this case,
-/// the `Sink` is only interested in `Definition 4`, and ignores `Definition 3`.
-///
-/// Finally, The merged definition (named `Definition 1 & 2`), and `Definition 4` are merged
-/// together to produce the new `Definition` returned by this method.
-pub fn merged_definition(
+/// The cache is used whilst building up the topology.
+/// TODO: Describe more, especially why we have a bool in the key.
```

Comment:
I'm not totally sure why. I think @fuchsnj had an idea?

Reply:
This was to separate items in the cache that were calculated with `schema_enabled` vs without. Since that's a global setting, this might not actually be necessary, but it does future proof this in case that ever does happen.